pulling out bbox fun and updating euro vax plot

mrajeev08 opened the pull request:
Hi Sean, 

Had a go at fixing a few things:

- wrapped text on 'fully' vaccination plot 
- pulled out the bbox code into a function (note did not document or test, just tagged as internal)
- for EURO: set ymax from 9500000 -> 820000
- also made background on legend transparent, this only really makes a difference for EURO, but only way I could think of doing without changing code to move legend.position depending on region (using scales::alpha, I think ok because scales is already a dependency)

For some reason, the fig looks a bit different than the other regions (no margins for some reason?)--not sure what's going on there.
![Screenshot 2022-05-10 175912](https://user-images.githubusercontent.com/42722106/167729444-f5dcefd7-27d7-4cb2-82cf-6bb80a53175a.png)

Not sure if I made it too much better, but atleast you can sort of see Portugal;)

mrajeev08 commented:
tried fixing bug in code by adding df to function arg (oof sorry Sean the git/github learning curve;)

beansrowning reviewed (changes requested):
Just a few things to look into, thanks for the quick work!

mrajeev08 commented on R/viz_maps.R:
```diff
@@ -322,3 +301,17 @@ map_vaccinations <- function(df, region = c("WHO Region", "State Region"), vac_t
 }
+
+#' @keywords internal
+bbox_fun <- function(who_region, df) {
```
Done, also added a little documentation (don't think it will show up in the pkgdown as tagged with @keyword internal?)

mrajeev08 commented:
Requested changes: 
- Moved bbox fun to R/lookups.R
  - Added a wee bit of documentation to it
  - Shifted the bbox of EURO ymin the same amount as previously done for ymax to fix plot
sizing
- Moved the legend transparency code to the map_template function
- Broke line in 'fully' plot at per 100 people

beansrowning commented on R/lookups.R:
```diff
@@ -164,3 +164,21 @@ datasource_lk <- list(
+#' @description Used internally in viz_maps.R functions for maps
+#' @returns a vector of class bbox (from the sf package)
+#'  with the min and max xy coordinates of the bounding box
+#' @keywords internal
```
Saw your comment RE: internal. That's fine with me, I think this is more documented than it was anyways

beansrowning reviewed (approved):
Nicely done! Ready to Merge.
